mnd/mnd/models.py

```diff
@@ -99,13 +99,33 @@ def clean(self):
 @receiver(user_email_updated)
 def update_carer_email(sender, user, previous_email, **kwargs):
     if user.is_carer:
-        if carer_patient := Patient.objects.filter(carer=user).first():
-            if primary_carer := carer_patient.primary_carers.first():
-                primary_carer.email = user.email  # Synchronise the carer's email with the user's updated email
+        new_email = user.email
+        for primary_carer in PrimaryCarer.objects.filter(email__iexact=previous_email):
+
+            # Get any deactivated registrations
+            deactivated_registrations = primary_carer.carerregistration_set.filter(status=CarerRegistration.DEACTIVATED)
+            # Get associated relationships
+            relationships_to_disassociate = PrimaryCarerRelationship.objects.filter(
+                carer=primary_carer,
+                patient__in=[r.patient for r in deactivated_registrations]
```

Comment:
It's unlikely that there will be that the loop will be huge given that the list of primary carers is limited to the one carer changing their email address, so I will look at using a transaction, because I don't think I can achieve the same thing with a bulk update? (e.g. creating a copy, modifying the copy, saving it).

Reply:
Made the triggering function atomic: https://github.com/eresearchqut/trrf/pull/615/files
Replaced list comprehension with subquery.